src/tools/rustfmt/src/expr.rs

```diff
@@ -225,6 +225,10 @@ pub(crate) fn format_expr(
         ast::ExprKind::Ret(Some(ref expr)) => {
             rewrite_unary_prefix(context, "return ", &**expr, shape)
         }
+        ast::ExprKind::Yeet(None) => Some("do yeet".to_owned()),
+        ast::ExprKind::Yeet(Some(ref expr)) => {
+            rewrite_unary_prefix(context, "do yeet ", &**expr, shape)
+        }
```

Comment:
Typically with a new variant we'd start with rustfmt just spitting back out whatever the user wrote, but in this particular case I can't imagine a universe where we'd do anything different. As such I've no concerns with these rustfmt changes :+1: 

Will use this to mention an aside, and realize the style team hasn't fully come to fruition yet @joshtriplett, but I think one future question/consideration should be when/how new syntax gets incorporated into the guide, including for cases like this that are obvious and noncontroversial.

Reply:
Oh, thanks @calebcartwright; I wasn't thinking about the stability details here at all.  But yes, I agree that I'd be *very* surprised if we wanted this to format differently from `return`/`break`.  Also, the syntax for this *will* change later, so we'll have another opportunity to rethink the formatting once that happens.

As another aside: Is there a way this `match` could have a catch-all arm that just does the "spit back what the user wrote", so that additions to the AST like this would do the right thing without needing a code change?  I only updated the file because the PR build broke when I did nothing.

Reply:
> Is there a way this match could have a catch-all arm that just does the "spit back what the user wrote", so that additions to the AST like this would do the right thing without needing a code change? 

Potentially. IIRC many of the big match expressions used to have a catch-all pattern arm in the earlier days when rustfmt used the AP crates, but that was intentionally changed some time back as part of an effort to ensure the rustfmt team would become aware of new syntax associated with new variants.

If/when we can implement some of the process changes previously discussed elsewhere then I'd certainly be open to adding such arms back. In the interim, for future reference, the behavior can be triggered with `None` or something like `Some(context.snippet(<<span>>).to_owned())`